### ISSUE TITLE
Problems and requests

### ISSUE THREAD

**Issue by GoldElysium** [jellyfin/jellyfin-plugin-anime#39]
**Issue 1**
AniDB pulls german? data. It should be english as I set that in settings. (#24)

**Issue 2**
Also, episode data doesn't work correctly. I have it now with Mahouka Koukou no Rettousei and Rakudai Kishi no Cavalry, that all episodes are named the same. The names with both were non existent episode names, with Mahouka it was `episode 162` as name and with Rakudai I can't remember but it wasn't correct either. Love Live! did work correctly tho?, everything was called the correct name with the correct description.
This issue is the most annoying for me, manually editting all the metadata is a pain in the ass.

**Request**
Add [Jikan](https://jikan.moe/), for MAL support. I use MAL alot, and prefer it above AniDB. Jikan is unofficial but it would be nice if it was added.
The [.Net](https://github.com/Ervie/jikan.net) version.

**Comment by GoldElysium** [comment]
As I have done C#/.Net before I **may** fork this and add Jikan myself and maybe solve the issues if I can find them.

**Comment by CTalvio** [comment]
I have not been able to get AniDB to work at all, it either falls back on TVDB or anisearch/proxer (both of which net data in german using this plugin, anisearch also exists in english but this plugin points to the .de version) 

Only TVDB or AniDb are supported for episode titles/metadata so the error is cause by some of your series being matched to TVDB correctly and some not. For it to work correctly the filenames seem to have to include season and episode number (most anime releases do not) and for them to correctly parsed, when they are not things get freaky and you get all the episodes detected as episode 1.

**Comment by ThatNerdyPikachu** [comment]
@CTalvio Can you see if the PR above at least solves your Anisearch issue?

**Comment by CTalvio** [comment]
@ThatNerdyPikachu Sorry, I don't have this figured out enough where I could take your changes all the way to a usable .dll. Provide that though and I'd be happy to test for you, not a dev of any kind myself, just here to be a datapoint when needed.

**Comment by Matt07211** [comment]
For issue 2 in your post this may be relevant, https://github.com/jellyfin/jellyfin-plugin-anime/issues/33

And as for issue 1 I would guess that since episode metadata isn't getting pulled its falling back to the German metadata providers.

As for MAL I use it a lot to, but until they add back an official API it doesn't seem like a good idea, https://github.com/jellyfin/jellyfin-plugin-anime/issues/7#issuecomment-457905844

**Comment by GoldElysium** [comment]
Ok, I understand that about MAL. I recently switched to Kitsu, still using MAL and anidb next to it. Kitsu has an [api](https://kitsu.docs.apiary.io/). So maybe this is an option to add? Most of the anime have episode descriptions and things like that, some are missing but if it keeps falling back to german this is better than nothing.

**Comment by Matt07211** [comment]
Yeah gotta wait for someone to implement it, I can certainly give it a go with no promises (mainly because I don't know what in doing and doing it all blind)

**Comment by GoldElysium** [comment]
@ThatNerdyPikachu 
Can't get it to build: 
`CS0117: 'HttpRequestOptions' does not contain a definition for 'ResourcePool'	Jellyfin.Plugin.Anime`
Haven't worked with .NET in a long time, could you maybe build it and send it as a download link?

**Comment by bobberb** [comment]
@ThatNerdyPikachu 

same build failure here.

**Comment by GoldElysium** [comment]
Looks like #41 got merged with release [v4](https://github.com/jellyfin/jellyfin-plugin-anime/releases/tag/v4), I'll check tomorrow if this fixes issue 1(and maybe also 2). I would still like to see a [Kitsu api](https://kitsu.docs.apiary.io/) implementation if possible, can someone try to code this?

**Comment by oddstr13** [comment]
Feature requests go here; https://features.jellyfin.org/
Makes it easy to vote and see what's popular.

**Comment by GoldElysium** [comment]
> Feature requests go here; https://features.jellyfin.org/
> Makes it easy to vote and see what's popular.

Oh ok, thanks for pointing that out!

**Comment by GoldElysium** [comment]
Hmm, anidb doesn't give **any** episode metadata, It doesn't give an description and I also still have the issue of that it's pulling the incorrect episode metadata.

**Comment by CTalvio** [comment]
@GoldElysium 
How are you testing the plugin? I have a separate library with files for the monogatari series (since that is one of the most difficult series to pull metadata for) and only AniDB enabled. Even with version 4 NO metadata is pulled from aniDB. The library remains blank with just folder names and filenames as entries.

My main libraries are sort functional with metadata from just TVDB and the few that work from this plugin. As far as I can tell AniDB agent is still completely nonfunctional.

Also, the episode names are because the JF is detecting them as all the same episode, not because of the plugin, you need to change the filenames to correct how they are being detected or do the file move dance (remove series from library, readd) to refresh them.

@zepx 
Is there something I am missing?

**Comment by GoldElysium** [comment]
> @GoldElysium
> How are you testing the plugin? I have a separate library with files for the monogatari series (since that is one of the most difficult series to pull metadata for) and only AniDB enabled. Even with version 4 NO metadata is pulled from aniDB. The library remains blank with just folder names and filenames as entries.
> 
> My main libraries are sort functional with metadata from just TVDB and the few that work from this plugin. As far as I can tell AniDB agent is still completely nonfunctional.
> 
> Also, the episode names are because the JF is detecting them as all the same episode, not because of the plugin, you need to change the filenames to correct how they are being detected or do the file move dance (remove series from library, readd) to refresh them.
> 
> @zepx
> Is there something I am missing?

I do it the same way you do, having only AniDB enabled. This is how two of my folders look(both aren't getting pulled correctly): 
![image](https://user-images.githubusercontent.com/48455312/62698714-17854900-b9de-11e9-93c5-8ed3194523f6.png) 
![image](https://user-images.githubusercontent.com/48455312/62698923-99757200-b9de-11e9-8df3-0da3775dc7ec.png)

I think this should be enough in the file name. How do you have your files named? 

Also, I posted the request on the site now. [link](https://features.jellyfin.org/posts/237/kitsu-api-anime-plugin)

**Comment by CTalvio** [comment]
@GoldElysium
All of my files follow the SxxExx naming convention, it seems to be the most reliably detected. I had tons of series doing the same thing you are describing that were simply numbered.

Your examples both fail to match the ones given in them filename guide. The only supported way to give a single digit for episode number is having it at the very start of the filename: https://web.archive.org/web/20181216120039/https://github.com/MediaBrowser/Wiki/wiki/TV-naming

This link is given when viewing the library management window, yes it's an archive link, this info is completely missing from the actual wiki.

Edit: note that if at any point you had TVDB enabled you won't be able to tell if new metadata is pulled from AniDB. Are you getting metadata populated on V4 on a blank library with just AniDB?

**Comment by zepx** [comment]
I just tested the plugin by creating a library from scratch. AniDB's metadata are indeed populating my Anime series. For starters, I suggest clearing the cache file created by the previous version of this plugin, then rescan your library.

`rm -rf <location of jellyfin>/cache/anidb`

The pull request solves a major issue with downloading a cache file anidb.

**Comment by CTalvio** [comment]
@zepx 
Brilliant! That did it, we will want to spread that info to users.
I remember this issue also appeared and was similarly solved with the HamaTV plugin for plex, thanks a bunch!!

Episode and season metadata seems to still be failing, filenames are still being displayed, though with this fix already be able to far better get my library together.

With this, might it be time to close this issue?

**Comment by zepx** [comment]
I will take a look at the episode and season metadata when I have time, I didn't realized that it's not pulling those correctly.

**Comment by GoldElysium** [comment]
> @GoldElysium
> All of my files follow the SxxExx naming convention, it seems to be the most reliably detected. I had tons of series doing the same thing you are describing that were simply numbered.
> 
> Your examples both fail to match the ones given in them filename guide. The only supported way to give a single digit for episode number is having it at the very start of the filename: https://web.archive.org/web/20181216120039/https://github.com/MediaBrowser/Wiki/wiki/TV-naming
> 
> This link is given when viewing the library management window, yes it's an archive link, this info is completely missing from the actual wiki.
> 
> Edit: note that if at any point you had TVDB enabled you won't be able to tell if new metadata is pulled from AniDB. Are you getting metadata populated on V4 on a blank library with just AniDB?

That did it, thanks! Yeah I'm doing it on an empty library. 



> @zepx
> Brilliant! That did it, we will want to spread that info to users.
> I remember this issue also appeared and was similarly solved with the HamaTV plugin for plex, thanks a bunch!!
> 
> Episode and season metadata seems to still be failing, filenames are still being displayed, though with this fix already be able to far better get my library together.
> 
> With this, might it be time to close this issue?

It's indeed only giving the filenames for me too

**Comment by ThatNerdyPikachu** [comment]
Bump, still an issue.
